pandas/core/reshape/pivot.py

```diff
@@ -368,18 +368,34 @@ def _convert_by(by):
 @Appender(_shared_docs['pivot'], indents=1)
 def pivot(data, index=None, columns=None, values=None):
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        if index is None:
+            cols = [columns]
+        else:
+            if is_list_like(index):
+                # If a given index is a list, set cols to index.
+                cols = index
+            else:
+                # If a given index is not a list, set cols to a list of index.
+                cols = [index]
+            cols.append(columns)
         append = index is None
         indexed = data.set_index(cols, append=append)
+
     else:
         if index is None:
-            index = data.index
+            index = MultiIndex.from_arrays([data.index, data[columns]])
+        elif is_list_like(index):
+            # Iterating through the list of multiple columns of an index.
+            indexes = [data[column] for column in index]
+            indexes.append(data[columns])
+            index = MultiIndex.from_arrays(indexes)
         else:
+            # Build multi-indexes if index is not None and not a list.
```

Comment:
can you make this an if/elseif/else

Reply:
I am not sure about the last if/else statement. I can pull if/elif/else for the first three but I still need to have another condition for the last two.